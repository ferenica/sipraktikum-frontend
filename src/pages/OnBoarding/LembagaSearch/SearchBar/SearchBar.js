import React from 'react';
import PropTypes from 'prop-types';

import {styled} from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import {ImSearch} from 'react-icons/im';

const StyledTextField = styled(TextField)({
  width: '80%',
  backgroundColor: 'white',
  margin: '8px',
});

const StyledInputAdornment = styled(InputAdornment)({
  height: '15px',
  alignItems: 'normal',
});

const SearchBar = (props) => {
  return (
    <StyledTextField
      id="search-bar"
      variant="outlined"
      placeholder='Cari nama lembaga'
      size="small"
      value={props.value}
      onChange={props.searchHandler}
      InputProps={{
        startAdornment: <StyledInputAdornment position="start">
          <ImSearch color='#FF8326'/>
        </StyledInputAdornment>,
      }}>
    </StyledTextField>
  );
};

export default SearchBar;

SearchBar.propTypes = {
  value: PropTypes.string,
  searchHandler: PropTypes.func,
};
