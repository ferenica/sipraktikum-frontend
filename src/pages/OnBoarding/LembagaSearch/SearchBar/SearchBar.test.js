import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import SearchBar from './SearchBar';

Enzyme.configure({adapter: new EnzymeAdapter()});

describe('Test SearchBar has components', () => {
  it('has', () => {
    const wrapper = mount(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
