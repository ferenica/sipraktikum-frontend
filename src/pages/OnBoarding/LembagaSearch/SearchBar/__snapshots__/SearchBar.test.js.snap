// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchBar has components has 1`] = `ReactWrapper {}`;
